### PR TITLE
added authentication

### PR DESCRIPTION
**Pull request by gsrai**
and login form and logout button, also removed admin link from the application template.

### CODE REVIEW

**Comment by gsrai** [comment]
@s-chiriac @lukehowkins try and hack into the admin page, i haven't tested it thoroughly, ill tell you the credentials to access the page tomorrow at work.

File: app/templates/application.hbs [hunk 1]
```diff
@@ -4,7 +4,6 @@
         {{#link-to 'ladder'}}Ladder{{/link-to}}
         {{#link-to 'rules'}}Rules{{/link-to}}
         {{#link-to 'history'}}History{{/link-to}}
-        {{#link-to 'admin'}}Admin{{/link-to}}
```
Comment (lukehowkins) [inline app/templates/application.hbs]: Could wrap this with the authenticated if statement as well.

Comment (gsrai) [inline app/templates/application.hbs]: why? only the admin page needs authenticating.

File: app/templates/admin.hbs [hunk 2]
```diff
@@ -1,92 +1,115 @@
...
-          </label>
-      </fieldset>
-      <fieldset>
-        <input type="submit" value="submit" class="submitButton">
-      </fieldset>
-  </form>
-</div>
-<div class="admin-container">
-  <h2>Add Game</h2>
+{{#if isAuthenticated}}
```
Comment (lukehowkins) [inline app/templates/admin.hbs]: Do all the tests still pass?

Comment (gsrai) [inline app/templates/admin.hbs]: not sure, might have some errors due to firebase not being defined as its new. i think the app is mvp ready

File: app/templates/application.hbs (hunk 1 above)
Comment (lukehowkins) [inline app/templates/application.hbs]: So that when logged in as admin you can switch between the pages easier. Just around the Admin link.

Comment (gsrai) [inline app/templates/application.hbs]: you can switch between pages as normal now, when admin page loads, the init function checks firebase if the user is already authenticated.

**Pull request merged**